ibis/backends/dask/core.py

```diff
@@ -110,20 +110,299 @@
 
 from __future__ import absolute_import
 
+import functools
 from typing import Optional
 
 import dask.dataframe as dd
+from multipledispatch import Dispatcher
 
+import ibis.common.exceptions as com
+import ibis.expr.operations as ops
+import ibis.expr.types as ir
+from ibis.backends.pandas import aggcontext as agg_ctx
 from ibis.backends.pandas.core import (
-    execute,
+    compute_time_context,
     is_computable_input,
     is_computable_input_arg,
 )
+from ibis.client import find_backends
+from ibis.expr.scope import Scope
+from ibis.expr.timecontext import canonicalize_context
 from ibis.expr.typing import TimeContext
 
+from .dispatch import execute_literal, execute_node, post_execute, pre_execute
+from .trace import trace
+
 is_computable_input.register(dd.core.Scalar)(is_computable_input_arg)
 
 
+def execute_with_scope(
```

Comment:
Can you refactor this to take a "execute function"? Then I think this can be shared between pandas/dask

Reply:
We don't have to do this, but I think it's worthwhile to try out this idea.


```
def main_execute(...):
      execute_dispatcher= ExecuteDispatcher()
      execute_dispatcher.register_execute_node(dask_execute_node)
      ....
      returm execute_with_scope(execute_dispatcher, op, ...)

```

Reply:
I think we should do this all as a separate refactor. There's a lot of commonalities between the pandas backend (dispatchers, traces, agg context in the future?) that I think will be easier to consolidate all in one go once the dask backend reaches feature parity.

Reply:
That's fine. Although I don't think we need to do everything in one go. Refactoring the "execute_*" methods are probably easy to get started since they are pretty much the same structure / copy paste code. 

Refactoring aggcontext could be a bit harder since we don't know if the structure matches there.

Either way, fine with closing this now.